src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -239,7 +309,7 @@
  * </p>
  * <pre>
  * &lt;module name=&quot;ImportOrder&quot;&gt;
- *   &lt;property name=&quot;groups&quot; value=&quot;*,javax,java&quot;/&gt;
+ *   &lt;property name=&quot;groups&quot; value=&quot;*,java,javax&quot;/&gt;
```

Comment:
Why was this changed?

Reply:
When i was testing this config with code example i was getting wrong order at line 329 i thought it was bcoz of top static import but it was also bcoz of this value ordered javax then java. So i could either change the order in code example Or here, i did it here.

Reply:
When i was testing this config with below code example i was getting wrong order violation at line 392 i thought it was only bcoz of above static import but it was also bcoz of this valued ordered javax then java so to correct either i change the imports or this value here. I changed here

Reply:
I think it would be better to keep `java, javax` and change the example; `java, javax` is the convention followed by many organizations and IDE's.  Remember, users often copy and paste directly from examples.

Reply:
ugh, if you keep` java then javax` then there is no need to change code.

Reply:
>  i was getting wrong order violation at line 392

line 392 of what file? I am confused by your comment. The code example on line 392 of `ImportOrderCheck.java` is not associated with this config.

Reply:
i am so sorry i wasted a lot of time it's not 392 it's 329 sorry

Reply:
Thanks for clarifying that.  Please come up with some way to keep `java, javax` in config but still show good violation examples.

Reply:
Thanks for clarifying, I was confused by "code example"; you were referring to the configuration. I see now.

Reply:
It's may be bcoz of my english sorry for that. 
So now should i make any changes here?

Reply:
I think this would be a good example that is representative of the intellij default import settings:

```bash

➜  src cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
        <module name="ImportOrder">
            <property name="groups" value="*,javax,java"/>
            <property name="ordered" value="true"/>
            <property name="separated" value="false"/>
            <property name="option" value="bottom"/>
            <property name="sortStaticImportsAlphabetically" value="true"/>
        </module>
        <module name="SuppressionXpathSingleFilter">
            <property name="checks" value="ImportOrder"/>
            <property name="message" value="^'java\..*'.*"/>
        </module>
    </module>
</module>

➜  src cat TestClass.java 
import static java.awt.Button.A;
import java.net.URL; // Violation, wrong order. Caused by static import above; all static imports should be at the bottom of all imports.
import java.security.KeyManagementException;

import javax.net.ssl.TrustManager; // Violation, wrong order and extra separation. 'javax' imports should be above 'java' imports and below 'type' import group with no separation.
import com.google.common.primitives.Doubles.BYTES; // Violation,  wrong order. This import should be above all static and 'java*' imports, in 'type' import group.
public class TestClass { }

➜  src java -jar checkstyle-8.39-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/Tester/src/TestClass.java:2:1: Wrong order for 'java.net.URL' import. [ImportOrder]
[ERROR] /home/nick/Desktop/Tester/src/TestClass.java:5:1: Extra separation in import group before 'javax.net.ssl.TrustManager' [ImportOrder]
[ERROR] /home/nick/Desktop/Tester/src/TestClass.java:5:1: Wrong order for 'javax.net.ssl.TrustManager' import. [ImportOrder]
[ERROR] /home/nick/Desktop/Tester/src/TestClass.java:6:1: Wrong order for 'com.google.common.primitives.Doubles.BYTES' import. [ImportOrder]
Audit done.
Checkstyle ends with 4 errors.

```
We cannot change the order of the imports in the config, since this config is made to emulate the import order of intellij.

Reply:
i have suggestion you have described a lot in violation comments this will give us error when we do `mvn clean verify` so i was thinking if we add/describe these as bulleted list above the configuration?

Reply:
I would say just line wrap and ignore the "separation" errors.

Reply:
how about this

```
import static java.awt.Button.A;
import java.net.URL; // Violation, wrong order. Caused by static import above;
import java.security.KeyManagementException;

import javax.swing.JComponent; // Violation, Extra separation and javax should be above java import
import com.neurologic.http.HttpClient; // Violation, should be above static and 'java*' imports

public class TestClass { }
```

Reply:
one last time! pls check this 

```
import static java.awt.Button.A;
import java.net.URL; // Violation, Wrong order, caused by static import above;

import java.security.KeyManagementException; // Violation, Extra separation
import javax.swing.JComponent; // Violation, Wrong order, 'javax' should be above 'java' imports
import com.neurologic.http.HttpClient; // Violation, Should be above static and 'java*' imports

public class TestClass { }
```

Reply:
Resolved.